packages/ketcher-react/src/script/editor/operations/text.ts

```diff
@@ -0,0 +1,189 @@
+
+    BaseOperation.invalidateItem(restruct, 'texts', this.data.id, 1)
+  }
+
+  invert(): BaseOperation {
+    return new TextDelete(this.data.id)
+  }
+}
+
+interface TextUpdateData {
+  id?: any
+  label?: string
+  invertedLabel?: string
+  position?: Vec2
+  type?: string
+}
+
+export class TextUpdate extends BaseOperation {
+  data: TextUpdateData
+
+  constructor(
+    id?: any,
+    label?: string,
+    invertedLabel?: string,
+    position?: Vec2,
+    type?: string
+  ) {
+    super(OperationType.TEXT_UPDATE)
+    this.data = { id, label, invertedLabel, position, type }
+  }
+
+  execute(restruct: Restruct) {
+    const { id, label } = this.data
+    const text = restruct.molecule.texts.get(id)
+
+    if (text) {
+      text.label = label
+    }
+
+    BaseOperation.invalidateItem(restruct, 'texts', id, 1)
+  }
+
+  invert() {
+    const inverted = new TextUpdate(this.data.id)
+    inverted.data.label = this.data.invertedLabel
+    inverted.data.invertedLabel = this.data.label
+    return inverted
+  }
+}
+
+interface TextDeleteData {
+  id: any
+  label?: string
+  position?: Vec2
+  type?: string
+}
+
+export class TextDelete extends BaseOperation {
+  data: TextDeleteData
+  performed: boolean
+
+  constructor(id: any) {
+    super(OperationType.TEXT_DELETE)
+    this.data = { id }
+    this.performed = false
+  }
+
+  execute(restruct: Restruct): void {
+    const struct = restruct.molecule
+
+    if (!this.performed) {
+      const item = struct.texts.get(this.data.id)!
+
+      this.data.label = item.label || ''
+      this.data.position = item.position
+      this.data.type = item.type
+      this.performed = true
+    }
+
+    restruct.markItemRemoved()
+    restruct.clearVisel(restruct.texts.get(this.data.id).visel)
+    restruct.texts.delete(this.data.id)
+
+    struct.texts.delete(this.data.id)
+  }
+
+  invert(): BaseOperation {
+    return new TextCreate(
+      this.data.id,
+      this.data.label,
+      // @ts-ignore
+      this.data.position,
+      this.data.type
+    )
+  }
+}
+
+interface TextMoveData {
+  id: any
+  d: any
+  noinvalidate: boolean
+}
+
+export class TextMove extends BaseOperation {
+  data: TextMoveData
+
+  constructor(id: any, d: any, noinvalidate: boolean) {
+    super(OperationType.TEXT_MOVE)
+    this.data = { id, d, noinvalidate }
+  }
+
+  execute(restruct: Restruct): void {
+    const struct = restruct.molecule
+    const id = this.data.id
+    const difference = this.data.d
+    const item = struct.texts.get(id)
+
+    item?.position?.add_(difference)
+    restruct.texts
+      .get(id)
+      .visel.translate(scale.obj2scaled(difference, restruct.render.options))
+    this.data.d = difference.negated()
+
+    if (!this.data.noinvalidate) {
+      BaseOperation.invalidateItem(restruct, 'texts', id, 1)
+    }
+  }
+
+  invert(): BaseOperation {
+    const move = new TextMove(this.data.id, this.data.d, this.data.noinvalidate)
+
+    move.data = this.data
+
+    return move
+  }
+}
```

Comment:
why do you didn't separate `TextUpdate` and `TextMove` by files?

Reply:
The length of the file does not exceed 200 lines, so it's not strictly necessary to split it into separate files for the individual operations